### DISCUSSION TITLE
Save the DetailList column width to db

### DISCUSSION THREAD

**Discussion by it950** [microsoft/fluentui#23850] [Q&A] [answered]
Can we save the changed column width of a DetailList to db?
eg:
Adjust a column width, and save the changes to db, then reload the page,  the column will be displayed with the saved width.

**Comment by spmonahan** [comment, marked as answer]
There are APIs on `DetailsList` that will allow you to do this. The actual implementation of how you save and load data from a database is outside of the scope of Fluent and you're likely to run into some edge cases (e.g., save the changes to DB, close the page, resize browser window width, reopen the page -- what should the column widths be?) but I can offer a rough outline of how to do this.

1. Set column widths with the [`calculatedWidth` property on the `IColumn` interface](https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist#IColumn). This allows you to directly set the width of columns. You'll get these values by loading them from your database.
2. Listen for the `onColumnResize` event on each column. Alternatively you can listen for this event directly on `DetailsList`. Use the value provided by this event to update the column value (i.e., save it to your database).

[This Codepen](https://codepen.io/seanms/pen/ExEyMKB) is a basic example of how you could accomplish saving and loading column widths from a database.

**Reply by it950** [reply]
Thank you for your reply @spmonahan.
Professional and perfect and detailed .
